src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensRangeHandler.cs

```diff
@@ -76,44 +66,7 @@ public SemanticTokensRangeHandler(IGlobalOptionService globalOptions)
                 includeSyntacticClassifications: context.Document.IsRazorDocument(),
                 cancellationToken).ConfigureAwait(false);
 
-            return new RoslynSemanticTokens { Data = tokensData, IsFinalized = isFinalized };
-        }
-
-        private async Task<bool> IsDataFinalizedAsync(Project project, CancellationToken cancellationToken)
-        {
-            // We use a combination of IsFullyLoaded + the completed project compilation as the metric
-            // for isFinalized. It may not be completely accurate but this is only a a temporary fix until
-            // workspace/semanticTokens/refresh is implemented.
-            lock (_lock)
-            {
-                if (_projectIdToCompilation.TryGetValue(project.Id, out var compilationTask) && compilationTask.IsCompleted)
-                {
-                    // We don't want to hang on to the compilation since this can be very expensive,
-                    // but we do want to mark the compilation as being successfully retrieved.
-                    if (compilationTask.Result is not null)
-                    {
-                        _projectIdToCompilation[project.Id] = Task.FromResult<Compilation?>(null);
-                    }
-
-                    return true;
-                }
-            }
-
-            var workspaceStatusService = project.Solution.Workspace.Services.GetRequiredService<IWorkspaceStatusService>();
-            var isFullyLoaded = await workspaceStatusService.IsFullyLoadedAsync(cancellationToken).ConfigureAwait(false);
-
-            lock (_lock)
-            {
-                if (isFullyLoaded && !_projectIdToCompilation.ContainsKey(project.Id))
-                {
-                    // If the project's compilation isn't yet available, kick off a task in the background to
-                    // hopefully make it available faster since we'll need it later to compute tokens.
-                    var newCompilationTask = project.GetCompilationAsync(cancellationToken);
```

Comment:
I assume we need this as well
https://github.com/dotnet/roslyn/pull/60991

Reply:
Actually, I don't think we need the linked PR. Razor already removed caching and the isFinalized logic from their end, so keeping it wouldn't have any effect